supplemental_notebooks/2020-07-29_dimensionality_reduction.Rmd

```diff
@@ -0,0 +1,237 @@
+---
+title: "Dimensionality Reduction concepts"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+author: Joshua Shapiro for CCDL
+date: 2020-07
+---
+
+
+```{r setup}
+library(tidyverse, quietly = TRUE)
+library(plotly) # 3D plotting
+
+#set seed for reproducibility
+set.seed(2020)
+```
+
+## What is "Dimensionality Reduction" anyway?
+
+We like to throw around the term "dimensionality reduction" a lot, but what is that? 
+In some ways, it is just what it says, reducing the number of dimensions in our data set, which is probably not very helpful.
+What do we mean by a dimension?
+
+A dimension, in this context, is a measurement that we have for all of our samples. 
+In the RNA-seq context, one dimension is the expression of a single gene.
+We have tens of thousands of genes that we have measured, so we have tens of thousands of dimensions. 
+
+Perhaps unsurprisingly, this can be very hard to visualize, as our brains have trouble thinking in more than 3 dimensions, and paper or computer screens effectively only show two at a time (maybe 3 with color and color vision).
+We have some techniques that are useful, such as heatmaps of the gene expression matrix, but these can be hard to interpret, and it is extremely hard to accurately show relationships among samples in a grid (hierarchical trees are useful, but also confusing and easily misinterpreted).
+
+So what we would like to do is take this information with all of its dimensions and reduce it down to something we can visualize, while maintaing as much of the underlying information as possible. 
+
+### Penguins
+
+To illustrate this, we will use a fun little data set called _Palmer Penguins_. This is contained in an R package may not be installed on your machine, so you can install it for your user with:
+
+```{r}
+install.packages("palmerpenguins")
+```
+
+We can then load the dataframe into our environment and look at its structure with:
+
+```{r}
+penguins <- palmerpenguins::penguins
+str(penguins)
+```
+
+What we have is a data frame (`tibble`, really) with 8 columns, so 8 dimensions!
+
+There is some missing data in there, so lets clean that up:
+```{r}
+penguins <- drop_na(penguins)
```

Comment:
I might add a comment about what this is doing. I don't remember what `drop_na()` does (same as `complete.cases()`? ) so I would appreciate the context, but maybe you don't want to get into it!

Reply:
Yeah, this way it is the same as `filter(complete.cases(.))`. It can also be applied to a single column: `drop_na(species)`